src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -117,27 +117,23 @@ public function process(ContainerBuilder $container)
             if (ChainAdapter::class === $class) {
                 $adapters = [];
                 foreach ($adapter->getArgument(0) as $provider => $adapter) {
-                    if ($adapter instanceof ChildDefinition) {
-                        $chainedPool = $adapter;
-                    } else {
+                    $chainedPool = $adapter;
+                    if (false === $adapter instanceof ChildDefinition) {
```

Comment:
use `if (!$adapter instanceof ...)` for consistency with the codebase.

Reply:
Fixed in 5859c93